### PR TITLE
`Dir.glob`: ignore Windows hidden files for `match_hidden: false`

### PR DESCRIPTION
**Pull request by HertzDevil**
Resolves the Windows part of #13196.

This specifically does not check for the system file attribute (`LibC::FILE_ATTRIBUTE_SYSTEM = 0x00000004`).

### CODE REVIEW

File: src/crystal/system/dir.cr [hunk 1]
```diff
@@ -4,17 +4,18 @@ module Crystal::System::Dir
...
   # If dir is nil, the type is unknown.
   # In the future we might change Dir's API to expose these entries
   # with more info but right now it's not necessary.
   struct Entry
     getter name
     getter? dir
+    getter? hidden
 
-    def initialize(@name : String, @dir : Bool?)
+    def initialize(@name : String, @dir : Bool?, @hidden : Bool)
```
Comment (Sija) [inline src/crystal/system/dir.cr:18]: btw, why `@dir` is nilable?

Comment (Sija) [reply src/crystal/system/dir.cr:18]: I don't get the reference.

Comment (straight-shoota) [reply src/crystal/system/dir.cr:18]: Line 10:
> # If dir is nil, the type is unknown.

Comment (HertzDevil) [reply src/crystal/system/dir.cr:18]: On POSIX a symlink is neither a (regular) file nor directory, you have to dereference it and hope that the target exists.

Comment (Sija) [reply src/crystal/system/dir.cr:18]: Thanks for clarification!

**Comment by straight-shoota** [comment]
I think we need to proceed with the discussion in #13196 to figure out the intended semantics and API.

**Pull request closed without merging**